### DISCUSSION TITLE
ViewField - pass data to view with accessible variable(s)

### DISCUSSION THREAD

**Discussion by joshembling** [filamentphp/filament#6338] [Help]
I can't find this in the docs and when I looked at the `ViewField` component in the source code, it didn't have any answers. 

I have a the following tab, which I have a custom view for. 

```
Tabs\Tab::make('Testing')
           ->schema([
                 ViewField::make('tests')
                  ->label(
                   function (?Model $record) {
                    return $record->relation;
                   }
                 )
                 ->view('test-results')
                 ->disabled(),
           ]),
```

How can access the $record, within the blade view, ideally with eager loaded relationships?

**Comment by danharrin** [comment]
`$getRecord()`

**Comment by joshembling** [comment]
Sorry I should have clarified further to the above - 

If I'm passing a json type field into the `ViewField::make('json_field')`, but I want to include certain keys of that object in the view, is it possible to chain something onto the ViewField to make it accessible in the view/change the default access?

**Reply by danharrin** [reply]
you would use a custom field class for that - view fields are only for simple use cases

**Reply by joshembling** [reply]
Can you point me in the right direction? Docs seem a little sparse in this area.

**Reply by danharrin** [reply]
https://ryangjchandler.co.uk/posts/building-a-custom-color-palette-field-in-filament-part-1

**Reply by danharrin** [reply]
`php artisan make:form-field` is the main command you need to create the class and the view

**Comment by photojo** [comment]
Is possible to pass data back from a ViewField:

  `ViewField::make('car')->view('filament.forms.components.cars')`

My filament.forms.components.cars contains a large Livewire-table with overview over a huge amount of cars. I want to fill 'car' by clicking one of these cars.
Is this possible?

Thank you.

**Reply by ArtMin96** [reply]
[Running actions](https://filamentphp.com/docs/3.x/tables/columns/getting-started#running-actions)?

**Reply by photojo** [reply]
Of course, I need an action.
So, let me show, what I have so far.
I have a filament-Table. I use Action::make .. and a form to edit rows. In the form I use ViewField to show a huge table (let us call it cars), made with livewire (no filament). 

 Action::make('edituser')
                    ->label('Edit User')
                    ->form([
                        TextInput::make('firstname'),
                        TextInput::make('userscar'),
                        ViewField::make('car')->view('filament.forms.fields.allcars')
                    ->action(...)

How can I fill the field 'userscar' by clicking on a car in the huge table, shown in the ViewField?

Thank you so much for any hints!

**Comment by photojo** [comment]
Any hints for me? I want to fill a field by clicking a dataset on a huge table.

**Reply by photojo** [reply]
So, I want to share my own solution:

1. Component with Filament Form
2. Livewire Component with (in my case) a Table
3. CustomField with alpine.js 
4. In Table: use x-on

```
<x-dynamic-component :component="$getFieldWrapperView()" :field="$field" :label="$getLabel()">
    <div x-data="{ selectedcar: $wire.$entangle('{{ $getStatePath() }}') }">
   <livewire:allcars key="{{ now() }}">
</div>

</x-dynamic-component>
```


In my table (all cars) I use ...x-on:click="selectedcar ='{{ $id }}'"...

Greetings 
Jo
